.github/scripts/setup_runners.sh

```diff
@@ -0,0 +1,120 @@
+  exit 1
+fi
+
+SETUP_ACTION=$1
+SETUP_INSTANCE=`echo $2 | sed 's/,/ /g'`
+SETUP_TOKEN=$3
+SETUP_AMI_ID="ami-0397210a1b7e7e74c"
+SETUP_AMI_USER="ec2-user"
+SETUP_INSTANCE_TYPE="m5.xlarge"
+SETUP_INSTANCE_SIZE=20 #GiB
+SETUP_KEYNAME="odfe-release-runner"
+SETUP_SECURITY_GROUP="odfe-release-runner"
+SETUP_IAM_NAME="odfe-release-runner"
+GIT_URL_API="https://api.github.com/repos"
+GIT_URL_BASE="https://github.com"
+GIT_URL_REPO="opendistro-for-elasticsearch/opendistro-build"
+
+echo "###############################################"
+echo "Start Running $0 $1 $2"
+echo "###############################################"
+
+###############################################
+# Run / Start instances and bootstrap runners #
+###############################################
+if [ "$SETUP_ACTION" = "run" ]
+then
+
+  echo ""
+  echo "Run / Start instances and bootstrap runners [${SETUP_INSTANCE}]"
+  echo ""
+
+  # Provision VMs
+  for instance_name1 in $SETUP_INSTANCE
+  do
+    echo "[${instance_name1}]: Start provisioning vm"
+    aws ec2 run-instances --image-id $SETUP_AMI_ID --count 1 --instance-type $SETUP_INSTANCE_TYPE \
+                          --block-device-mapping DeviceName=/dev/xvda,Ebs={VolumeSize=$SETUP_INSTANCE_SIZE} \
+                          --key-name $SETUP_KEYNAME --security-groups $SETUP_SECURITY_GROUP \
+                          --iam-instance-profile Name=$SETUP_IAM_NAME \
+                          --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=$instance_name1}]" > /dev/null 2>&1; echo $?
+    sleep 1
+  done
+
+  echo ""
+  echo "Sleep for 120 seconds for EC2 instances to start running"
+  echo ""
+
+  sleep 120
+
+  # Setup VMs to register as runners
+  for instance_name2 in $SETUP_INSTANCE
+  do
+    echo "[${instance_name2}]: Make change of the runner hostname"
+    aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo hostnamectl set-hostname '${instance_name2}'"]}' \
+                         --output text > /dev/null 2>&1; echo $?
+
+    echo "[${instance_name2}]: Get runner token and bootstrap on Git"
+    instance_runner_token=`curl --silent -H "Authorization: token ${SETUP_TOKEN}" --request POST "${GIT_URL_API}/${GIT_URL_REPO}/actions/runners/registration-token" | jq -r .token`
+    aws ssm send-command --targets Key=tag:Name,Values=$instance_name2 --document-name "AWS-RunShellScript" \
+                         --parameters '{"commands": ["#!/bin/bash", "sudo su - '${SETUP_AMI_USER}' -c \"cd actions-runner && ./config.sh --unattended --url '${GIT_URL_BASE}/${GIT_URL_REPO}' --labels '${instance_name2}' --token '${instance_runner_token}' && nohup ./run.sh &\""]}' \
+                         --output text > /dev/null 2>&1; echo $?
+    sleep 5
+  done
+
+  echo ""
+  echo "Wait for 60 seconds for runners to bootstrap on Git"
```

Comment:
same as above. not sure if github provides api to check the runner status

Reply:
Same as above